### ISSUE TITLE
R-devel: robustSmoothSpline() gives an error

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/aroma.light#9]
Package test `tests/robustSmoothSpline.R` for `robustSmoothSpline()` gives an error (occassionally) in R devel;

``` r
> library("aroma.light")
> data("cars")
> fit <- robustSmoothSpline(cars$speed, cars$dist)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
Error in smooth.spline0(spline.prep, g$yin, w = w, ...) : 
  smoothing parameter value too small (compared to 0.5): spar=-0.354101966249684 (caught because ier = 1 > 0.5)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
Error in smooth.spline0(spline.prep, g$yin, w = w, ...) : 
  smoothing parameter value too small (compared to 0.5): spar=-0.354101966249684 (caught because ier = 1 > 0.5)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
> fit <- robustSmoothSpline(cars$speed, cars$dist)
Error in smooth.spline0(spline.prep, g$yin, w = w, ...) : 
  smoothing parameter value too small (compared to 0.5): spar=-0.354101966249684 (caught because ier = 1 > 0.5)
```

**Comment by HenrikBengtsson** [comment]
Troubleshooting:  The error started to occur _after_ R-devel (2016-05-27 r70677):

``` sh
$ R-3.4.0devel-20160528/bin/Rscript tests/robustSmoothSpline.R 
aroma.light v3.3.1-9000 (2016-08-10) successfully loaded. See ?aroma.light for help.
$
```

and before or including R-devel (2016-05-28 r70682):

``` sh
$ R-3.4.0devel-20160530/bin/Rscript tests/robustSmoothSpline.R 
aroma.light v3.3.1-9000 (2016-08-10) successfully loaded. See ?aroma.light for help.
Error in smooth.spline0(spline.prep, g$yin, w = w, ...) : 
  smoothing parameter value too small (compared to 0.5): spar=-0.354101966249684 (caught because ier = 1 > 0.5)
Calls: robustSmoothSpline -> robustSmoothSpline.default -> smooth.spline0
Execution halted
$
```

**Comment by HenrikBengtsson** [comment]
In revisions (r70677, r70682], there are two commits of interest:

``` sh
$ svn log -r 70678:70682
r70678 | maechler | 2016-05-28 03:41:15 -0700 (Sat, 28 May 2016) | 2 lines
hatvalues.lm: names(.) are already there
[...]
r70682 | maechler | 2016-05-28 16:29:39 -0700 (Sat, 28 May 2016) | 1 line
smooth.spline() option spar=c(lamdbda = *); hatvalues()
```

The code updates are:
- [r70678](https://github.com/wch/r-source/commit/beb32b167756cd3e2b7d64000686b50ec637f1da)
- [r70682](https://github.com/wch/r-source/commit/d114e356c1c4941c175a97d5cc48ca21ab5b1119)

# Full log

``` sh
$ svn log -r 70677:70682
------------------------------------------------------------------------
r70677 | luke | 2016-05-27 14:14:15 -0700 (Fri, 27 May 2016) | 3 lines

Update on documentation of 'local' compilation rule from Tomas
Kalibera, with some mods.

------------------------------------------------------------------------
r70678 | maechler | 2016-05-28 03:41:15 -0700 (Sat, 28 May 2016) | 2 lines

hatvalues.lm: names(.) are already there

------------------------------------------------------------------------
r70679 | ligges | 2016-05-28 07:47:54 -0700 (Sat, 28 May 2016) | 1 line

add sanity checks to system2() on Windows that the Unix version already got
------------------------------------------------------------------------
r70680 | hornik | 2016-05-28 12:06:07 -0700 (Sat, 28 May 2016) | 1 line

Tweaks.
------------------------------------------------------------------------
r70681 | hornik | 2016-05-28 12:06:27 -0700 (Sat, 28 May 2016) | 1 line

Spelling.
------------------------------------------------------------------------
r70682 | maechler | 2016-05-28 16:29:39 -0700 (Sat, 28 May 2016) | 1 line

smooth.spline() option spar=c(lamdbda = *); hatvalues()
```

**Comment by HenrikBengtsson** [comment]
Fixed in commit 2686d18.
